service/matching/handler/engine.go

```diff
@@ -198,6 +198,24 @@ func (e *matchingEngineImpl) getTaskListManager(taskList *tasklist.Identifier, t
 	}
 	e.taskListsLock.RUnlock()
 
+	// Defensive check to make sure we actually own the task list
+	//   If we try to create a task list manager for a task list that is not owned by us, return an error
+	//   The new task list manager will steel the task list from the current owner, which should only happen if
+	//   the task list is owned by the current host.
+	tasklistOwner, err := e.membershipResolver.Lookup(service.Matching, taskList.GetName())
+	if err != nil {
+		return nil, fmt.Errorf("failed to lookup task list owner: %w", err)
+	}
+
+	self, err := e.membershipResolver.WhoAmI()
+	if err != nil {
+		return nil, fmt.Errorf("failed to lookup self im membership: %w", err)
+	}
+
+	if tasklistOwner.Identity() != self.Identity() {
+		return nil, errors.New("task list is not owned by this host")
```

Comment:
I would suggest returning a typed error so we can match off this in other parts of the system

Reply:
Fixed, we now return a typed error